models/models_methods_test.go

```diff
@@ -1193,175 +1193,101 @@ func TestGetStepIDStepDataPair(t *testing.T) {
-
-		require.NoError(t, err)
-		require.Equal(t, "path", stepIDData.SteplibSource)
-		require.Equal(t, "$HOME/some/path/in/home", stepIDData.IDorURI)
-		require.Equal(t, "", stepIDData.Version)
-	}
-
-	//
-	// ----- Direct git uri
-	t.Log("direct git uri")
-	{
-		stepCompositeIDString := "git::https://github.com/bitrise-io/steps-timestamp.git@develop"
-		stepIDData, err := CreateStepIDDataFromString(stepCompositeIDString, "some-def-coll")
-
-		require.NoError(t, err)
-		require.Equal(t, "git", stepIDData.SteplibSource)
-		require.Equal(t, "https://github.com/bitrise-io/steps-timestamp.git", stepIDData.IDorURI)
-		require.Equal(t, "develop", stepIDData.Version)
-	}
-
-	t.Log("direct git uri")
-	{
-		stepCompositeIDString := "git::git@github.com:bitrise-io/steps-timestamp.git@develop"
-		stepIDData, err := CreateStepIDDataFromString(stepCompositeIDString, "")
-
-		require.NoError(t, err)
-		require.Equal(t, "git", stepIDData.SteplibSource)
-		require.Equal(t, "git@github.com:bitrise-io/steps-timestamp.git", stepIDData.IDorURI)
-		require.Equal(t, "develop", stepIDData.Version)
-	}
-
-	t.Log("direct git uri")
-	{
-		stepCompositeIDString := "git::https://github.com/bitrise-io/steps-timestamp.git"
-		stepIDData, err := CreateStepIDDataFromString(stepCompositeIDString, "some-def-coll")
-
-		require.NoError(t, err)
-		require.Equal(t, "git", stepIDData.SteplibSource)
-		require.Equal(t, "https://github.com/bitrise-io/steps-timestamp.git", stepIDData.IDorURI)
-		require.Equal(t, "master", stepIDData.Version)
-	}
+	for _, tt := range []struct {
+		composite   string
+		stepSrc     string
+		wantStepSrc string
+		wantStepID  string
+		wantVersion string
+		wantErr     bool
+		name        string
+	}{
+		{
+			composite: "steplib-src::step-id@0.0.1", stepSrc: "",
+			wantStepSrc: "steplib-src", wantVersion: "0.0.1", wantStepID: "step-id", wantErr: false,
+			name: "default / long / verbose ID mode",
+		},
+		{
+			composite: "step-id@0.0.1", stepSrc: "default-steplib-src",
+			wantStepSrc: "default-steplib-src", wantVersion: "0.0.1", wantStepID: "step-id", wantErr: false,
+			name: "no steplib-source",
+		},
+		{
+			composite: "::step-id@0.0.1", stepSrc: "default-steplib-src",
+			wantStepSrc: "default-steplib-src", wantVersion: "0.0.1", wantStepID: "step-id", wantErr: false,
+			name: "invalid/empty step lib source, but default provided",
+		},
+		{
+			composite: "::step-id@0.0.1", stepSrc: "",
+			wantStepSrc: "", wantVersion: "", wantStepID: "", wantErr: true,
+			name: "invalid/empty step lib source + no default",
+		},
+		{
+			composite: "step-id@0.0.1", stepSrc: "",
+			wantStepSrc: "", wantVersion: "", wantStepID: "", wantErr: true,
+			name: "no steplib-source & no default -> fail",
+		},
+		{
+			composite: "step-id", stepSrc: "def-lib-src",
+			wantStepSrc: "def-lib-src", wantVersion: "", wantStepID: "step-id", wantErr: false,
+			name: "no steplib & no version, only step-id",
+		},
+		{
+			composite: "", stepSrc: "def-step-src",
```

Comment:
clear `def-step-src` and unify it, sometimes you use: def-step-src, sometimes: default-steplib-src, use a short stepSrc, like (StepLib)